core/store/migrations/migration1597695690/migrate.go

```diff
@@ -0,0 +1,194 @@
+package migration1597695690
+
+import (
+	"github.com/jinzhu/gorm"
+)
+
+// Migrate creates the offchain_reporting_job_specs table
+func Migrate(tx *gorm.DB) error {
+	return tx.Exec(`
+        CREATE TABLE pipeline_specs (
+            id SERIAL PRIMARY KEY,
+            dot_dag_source TEXT NOT NULL,
+            created_at timestamptz NOT NULL
+        );
+
+        CREATE INDEX idx_pipeline_specs_created_at ON pipeline_specs USING BRIN (created_at);
+
+        CREATE TABLE pipeline_task_specs (
+            id SERIAL PRIMARY KEY,
+            dot_id TEXT,
+            pipeline_spec_id INT NOT NULL REFERENCES pipeline_specs (id) ON DELETE CASCADE,
+            type TEXT NOT NULL,
+            json jsonb NOT NULL,
+            index INT NOT NULL DEFAULT 0,
```

Comment:
The `index` param is just me thinking forward to multi-word transactions.  We'll need to be able to specify the index of each final output of the pipeline so that know how to order them in the tx.

Reply:
Also fwiw we might come up with individual task types which, unlike `median`, will care about the ordering of their inputs